src/main/metabase/api.ts

```diff
@@ -286,3 +308,141 @@ export async function syncMetabaseSchema(): Promise<void> {
   }
   await post(`/api/database/${anonymizedAccessDbId}/sync`, {});
 }
+
+type CollectionId = number;
+
+async function createCollection(parentId: number, name: string, description?: string): Promise<CollectionId> {
+  const { id } = await post('/api/collection', {
+    name,
+    description: description || null,
+    color: '#509EE3',
+    parent_id: parentId,
+  });
+
+  return id as CollectionId;
+}
+
+type Collection = {
+  id: number;
+  personal_owner_id: number;
+  name: string;
+  archived: boolean;
+  children: Collection[];
+};
+
+// We can query the user ID from /api/user/current, but the initial user we create will always be 1.
+const ADMIN_USER_ID = 1;
+const EXAMPLES_COLLECTION_NAME = 'Examples';
+
+async function getExamplesCollection(): Promise<{
+  id: number;
+  children: Collection[];
+}> {
+  const collections = (await get('/api/collection/tree?tree=true')) as unknown as Collection[];
+  const personalCollection = find(collections, { archived: false, personal_owner_id: ADMIN_USER_ID });
+  if (!personalCollection) {
+    throw new Error('No personal collection found.');
+  }
+
+  const examplesCollection = find(personalCollection.children, { archived: false, name: EXAMPLES_COLLECTION_NAME });
+  if (examplesCollection) {
+    return examplesCollection;
+  } else {
+    return {
+      id: await createCollection(personalCollection.id, EXAMPLES_COLLECTION_NAME),
+      children: [],
+    };
+  }
+}
+
+async function addQueryToCollection(query: ExampleQuery, databaseId: number, collectionId: number): Promise<number> {
+  const response = await post('/api/card', {
+    name: query.name,
+    dataset_query: {
+      type: 'native',
+      native: {
+        query: query.sql,
+        'template-tags': {},
+      },
+      database: databaseId,
+    },
+    display: query.display,
+    description: null,
+    visualization_settings: query.visualizationSettings,
+    parameters: [],
+    collection_id: collectionId,
+    result_metadata: null,
+  });
+
+  console.log(response);
```

Comment:
maybe log a `"Response from POST /api/card", response` so that it is later more easier to find

Reply:
This should not be here! Forgot to remove it. Will force push.